backpack/extensions/secondorder/hbp/linear.py

```diff
@@ -61,3 +64,19 @@ def __mean_input_outer(self, module):
         N = module.input0.size(0)
         flat_input = module.input0.reshape(N, -1)
         return einsum("ni,nj->ij", (flat_input, flat_input)) / N
+
+    def check_parameters(self, ext, module: Linear) -> None:
+        """Raise an exception if module parameters are not supported.
+
+        Args:
+            ext (KFAC or KFRA or KFLR): Extension calling out to the module.
+            module: Linear layer.
```

Comment:
```suggestion
    def check_parameters(self, module: Linear) -> None:
        """Raise an exception if module parameters are not supported.

        Args:
            module: Linear layer.
```
I've been using this interface inside the ParameterDerivatives classes.
1. Is it supposed to be inside the extension?
2. Does it need the extension as argument?

Reply:
1. Yes, these extensions cannot fully be expressed as derivatives. So it is the extension that currently does not support certain hyperparameters.
2. The extension is currently passed to provide a more meaningful error message. Open for alternatives.